File: src/main/java/esport/sprint1/sprint1/models/Journaliste.java
package esport.sprint1.sprint1.models;

import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Entity
public class Journaliste extends Utilisateur implements Serializable {
    @OneToMany(mappedBy = "m_Writer")
    private List<Actualite> m_Acts;

    public List<Actualite> getM_Acts() {
        return m_Acts;
    }

    public Journaliste(List<Actualite> m_Acts) {
        this.m_Acts = m_Acts;
    }

    public Journaliste(String nom, String prenom, Long villeId, String nomUtilisateur, String email, String adresse, String pays, int codePostale, String apropos, String citation, int levele, int points, boolean newUser, boolean active, boolean confirm, String photo, String coverImg, Date created_at, Date updated_at, List<Actualite> m_Acts) {
        super(nom, prenom, villeId, nomUtilisateur, email, adresse, pays, codePostale, apropos, citation, levele, points, newUser, active, confirm, photo, coverImg, created_at, updated_at);
        this.m_Acts = m_Acts;
    }
    public Journaliste(){}
    public void setM_Acts(List<Actualite> m_Acts) {
        this.m_Acts = m_Acts;
    }
}

File: src/main/frontend/src/models/Journaliste.ts
import {Tournoi} from "./Tournoi";

export class Journaliste{
  id: number = null;
  organisateur: string = "";
  actualites: Array<Tournoi> = new Array();
  nom: string = "";
  prenom: string = "";

  villeId: number = null;

  nomUtilisateur: string = "";
  email: string = "";
  adresse: string = "";
  pays: string = "";
  codePostale: number = null;
  apropos: string = "";
  citation: string = "";

  levele: number = 0;
  points: number = 0;

  newUser: boolean = true;
  active: boolean = false;
  confirm: boolean = false;

  photo: string = "";
  coverImg: string = "";

  created_at: Date = new Date;
  updated_at: Date = null;

}
